logger.go

```diff
@@ -162,6 +162,11 @@ func (sumoLogger *sumoLogger) sendLogs(logs []*sumoLog) error {
   if sumoLogger.gzipCompression {
     request.Header.Add("Content-Encoding", "gzip")
   }
+  if sumoLogger.sourceCategory != "" {
```

Comment:
Does `!= ""` also handle `nil`/null?

Reply:
That's because in golang the string cannot be nil (but `*string` can)